charts/opentelemetry-collector/templates/servicemonitor.yaml

```diff
@@ -3,12 +3,22 @@ apiVersion: monitoring.coreos.com/v1
 kind: ServiceMonitor
 metadata:
   name: {{ include "opentelemetry-collector.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
```

Comment:
I don't think it's a good practice to install any components of the helm chart in another namespace. Do you have any example where this approach is being used?

Also why not just create the service monitor separately from the otel helm chart?

Reply:
The otel helm chart already gives you the opportunity to create the `serviceMonitor` and also a `podMonitor` which both components are from Prometheus. In fact if you try to install the otel helm chart without the Prometheus CRDs, specifying the serviceMonitor values you'll get the next error::

```
Error: INSTALLATION FAILED: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
```

 I agree with you that maybe these resources should be created separately from the otel helm chart, but since the the option was already in it, I only wanted to extend the options to configure the serviceMonitor due to the case that if you already have a Prometheus instance in a specific namespace this will help you to avoid create a new prometheus resources to scrape the metrics from the otel instance. As you may know in Prometheus you need to specify the `serviceMonitorSelector`.

Reply:
Thank you very much @Allex1 and @dmitryax for your comments, I'll proceed to close this PR.